test/unittests/t_catalog_mgr.cc

```diff
@@ -0,0 +1,237 @@
+    ASSERT_EQ(1u, root_catalog->GetChildren().size());
+    catalog_mgr_.RegisterNewCatalog(new_catalog);
+    // adding "/dir/dir/dir/file3" to the new nested catalog
+    hash = shash::Any(shash::kSha1,
+                      reinterpret_cast<const unsigned char*>(hashes[2]),
+                      suffix);
+    new_catalog->AddFile(hash, file_size, "/dir/dir/dir", "file3");
+    // adding "/dir/dir/dir/file4" to the new nested catalog
+    hash = shash::Any(shash::kSha1,
+                      reinterpret_cast<const unsigned char*>(hashes[3]),
+                      suffix);
+    new_catalog->AddFile(hash, file_size, "/dir/dir/dir", "file4");
+    // adding "/dir/dir/dir/dir"
+    new_catalog->AddFile(empty_content, file_size, "/dir/dir/dir", "dir");
+    // adding "/dir/dir/dir/dir/dir"
+    new_catalog->AddFile(empty_content, file_size, "/dir/dir/dir/dir", "dir");
+    // we haven't mounted the second catalog yet!
+    ASSERT_EQ(1, catalog_mgr_.GetNumCatalogs());
+
+    // mounting a new catalog in "/dir/dir/dir/dir/dir"
+    MockCatalog *new_catalog_2 = new MockCatalog("/dir/dir/dir/dir/dir",
+                                                 shash::Any(),
+                                                 4096, 1, 0, false,
+                                                 new_catalog, NULL);
+    ASSERT_EQ(1u, root_catalog->GetChildren().size());
+    catalog_mgr_.RegisterNewCatalog(new_catalog_2);
+    // adding "/dir/dir/dir/dir/dir/file5"
+    hash = shash::Any(shash::kSha1,
+                      reinterpret_cast<const unsigned char*>(hashes[4]),
+                      suffix);
+    new_catalog_2->AddFile(hash, file_size, "/dir/dir/dir/dir/dir", "file5");
+    // we haven't mounted the third catalog yet!
+    ASSERT_EQ(1, catalog_mgr_.GetNumCatalogs());
+  }
+
+ protected:
+  static const char *hashes[];
+  perf::Statistics statistics_;
+  MockCatalogManager catalog_mgr_;
+};
+
+const char *T_CatalogManager::hashes[] = {
+     "b026324c6904b2a9cb4b88d6d61c81d1000000",
+     "26ab0db90d72e28ad0ba1e22ee510510000000",
+     "6d7fce9fee471194aa8b5b6e47267f03000000",
+     "48a24b70a0b376535542b996af517398000000",
+     "1dcca23355272056f04fe8bf20edfce0000000"
+};
+
+
+
+TEST_F(T_CatalogManager, InitialConfiguration) {
+  EXPECT_TRUE(catalog_mgr_.Init());
+  EXPECT_EQ(1, catalog_mgr_.GetNumCatalogs());
+  EXPECT_EQ(1u, catalog_mgr_.GetRevision());
+  EXPECT_FALSE(catalog_mgr_.GetVolatileFlag());
+  EXPECT_EQ(0u, catalog_mgr_.GetTTL());
+}
+
+TEST_F(T_CatalogManager, Statistics) {
+  EXPECT_TRUE(catalog_mgr_.Init());
+  const Statistics &st = catalog_mgr_.statistics();
+  EXPECT_EQ(0u, st.n_listing->Get());
+  EXPECT_EQ(0u, st.n_lookup_inode->Get());
+  EXPECT_EQ(0u, st.n_lookup_path->Get());
+  EXPECT_EQ(0u, st.n_lookup_path_negative->Get());
+  EXPECT_EQ(0u, st.n_lookup_xattrs->Get());
+  EXPECT_EQ(0u, st.n_nested_listing->Get());
+}
+
+TEST_F(T_CatalogManager, InodeConfiguration) {
```

Comment:
Here it would be nice to add some code to actually test if the inode annotation works as expected